File: multipleReading.c
#include <stdio.h>
#include <string.h>

int main(){

    char entrada[200];

    scanf ("%s", entrada);

    int quantidadeLeiturasPorClocks=0;

    scanf ("%d", &quantidadeLeiturasPorClocks);

    int quantidadeClocks=0;
    int quantidadeLeituras=0;

    for (int i=0;i<strlen(entrada);i++){

        if (entrada[i]=='W'){

            quantidadeClocks++;

        }

        if (entrada[i]=='R'){

            quantidadeLeituras++;
        }
    }

    int aux= quantidadeLeituras/quantidadeLeiturasPorClocks;

    if (aux !=0){

        aux++;
    }

    quantidadeClocks+=aux;

    printf ("%d", quantidadeClocks);



    return 0;
}

File: MultipleReading.java

/* 
Isso ocorre porque temos quatro operações de leitura, e podemos processar todas elas simultaneamente,
 já que temos oito processos disponíveis. Portanto, apenas um ciclo de clock é necessário para executar
  todas as operações de leitura. 
*/
import java.util.Scanner;

public class MultipleReading{


    public static void main(String[] args){

        Scanner scanner= new Scanner (System.in);

        String entrada=scanner.nextLine();
        int qntdLeitura= scanner.nextInt();
        int quantidadeClocks=0;
        int quantidadeLeitura=0;
        for (int i=0;i<entrada.length();i++){

            if (entrada.charAt(i)=='W'){

                quantidadeClocks++;
            }

            if (entrada.charAt(i)=='R'){

                quantidadeLeitura++;
            }
        }

        int aux= quantidadeLeitura/qntdLeitura;

        if (quantidadeLeitura % qntdLeitura != 0) {
            aux++;
        }

        quantidadeClocks+= aux;

        
        System.out.println(quantidadeClocks);

        scanner.close();
    }

}